flow/e2e/peer_flow_test.go

```diff
@@ -351,3 +353,305 @@ func (s *E2EPeerFlowTestSuite) Test_Complete_Simple_Flow() {
 
 	env.AssertExpectations(s.T())
 }
+
+func (s *E2EPeerFlowTestSuite) Test_Toast() {
+	env := s.NewTestWorkflowEnvironment()
+	registerWorkflowsAndActivities(env)
+
+	_, err := s.pool.Exec(context.Background(), `
+	
+		CREATE TABLE e2e_test.test_toast (
+			id SERIAL PRIMARY KEY,
+			t1 text,
+			t2 text,
+			k int
+		);CREATE OR REPLACE FUNCTION random_string( int ) RETURNS TEXT as $$
+		SELECT string_agg(substring('0123456789bcdfghjkmnpqrstvwxyz', 
+		round(random() * 30)::integer, 1), '') FROM generate_series(1, $1);
+		$$ language sql;
+	`)
+	s.NoError(err)
+
+	connectionGen := FlowConnectionGenerationConfig{
+		FlowJobName:      "test_toast",
+		TableNameMapping: map[string]string{"e2e_test.test_toast": "test_toast"},
+		PostgresPort:     postgresPort,
+		BigQueryConfig:   s.bqHelper.Config,
+	}
+
+	flowConnConfig, err := connectionGen.GenerateFlowConnectionConfigs()
+	s.NoError(err)
+
+	env.OnActivity("FetchConfig", mock.Anything, mock.Anything).Return(flowConnConfig, nil)
+
+	peerFlowInput := peerflow.PeerFlowWorkflowInput{
+		PeerFlowName:   connectionGen.FlowJobName,
+		CatalogJdbcURL: postgresJdbcURL,
+		TotalSyncFlows: 1,
+		MaxBatchSize:   100,
+	}
+
+	// in a separate goroutine, wait for PeerFlowStatusQuery to finish setup
+	// and execute a transaction touching toast columns
+	go func() {
+		// wait for PeerFlowStatusQuery to finish setup
+		// sleep for 5 second to allow the workflow to start
+		time.Sleep(5 * time.Second)
+		for {
+			response, err := env.QueryWorkflow(
+				peerflow.PeerFlowStatusQuery,
+				connectionGen.FlowJobName,
+			)
+			if err == nil {
+				var state peerflow.PeerFlowState
+				err = response.Get(&state)
+				s.NoError(err)
+
+				if state.SetupComplete {
+					fmt.Println("query indicates setup is complete")
+					break
+				}
+			} else {
+				// log the error for informational purposes
+				fmt.Println(err)
+			}
+			time.Sleep(1 * time.Second)
+		}
+
+		/*
+			Executing a transaction which
+			1. changes both toast column
+			2. changes no toast column
+			2. changes 1 toast column
+		*/
+		_, err = s.pool.Exec(context.Background(), `
```

Comment:
Can we add more tests please?

Reply:
comparing table contents: https://github.com/PeerDB-io/peerdb/blob/main/flow/e2e/qrep_flow_test.go#L153